endtoend/components/validator.go

```diff
@@ -19,18 +19,164 @@ import (
 	contracts "github.com/prysmaticlabs/prysm/contracts/deposit-contract"
 	"github.com/prysmaticlabs/prysm/endtoend/helpers"
 	e2e "github.com/prysmaticlabs/prysm/endtoend/params"
-	"github.com/prysmaticlabs/prysm/endtoend/types"
+	e2etypes "github.com/prysmaticlabs/prysm/endtoend/types"
 	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 	"github.com/prysmaticlabs/prysm/shared/featureconfig"
 	"github.com/prysmaticlabs/prysm/shared/params"
 	"github.com/prysmaticlabs/prysm/shared/testutil"
+	"golang.org/x/sync/errgroup"
 )
 
 const depositGasLimit = 4000000
 
+var _ e2etypes.ComponentRunner = (*ValidatorNode)(nil)
+var _ e2etypes.ComponentRunner = (*ValidatorNodes)(nil)
+
+// ValidatorNodes represents set of validator nodes.
+type ValidatorNodes struct {
+	e2etypes.ComponentRunner
+	config  *e2etypes.E2EConfig
+	started chan struct{}
+}
+
+// NewValidatorNodes creates and returns a set of validator nodes.
+func NewValidatorNodes(config *e2etypes.E2EConfig) *ValidatorNodes {
```

Comment:
That's good point, and if you look at comments/description you'll notice that I refer to validator nodes as a set, since it was the naming I've originally used. Will revert to that `ValidatorNodeSet` terminology, since it is two of us know, who thinks that it will be a good idea.